File: src/types/artifacts.ts
export interface Artifact {
  id: string;
  kind: "image" | "output" | "checkpoint" | "dataset";
  path_type: "file" | "directory";
  display_name: string;
  content_url: string;
}
export interface DirectoryItem {
  display_name: string;
}
export interface DirectoryListing {
  items: DirectoryItem[];
}
export interface UploadImagesRequest {
  files: File[];
}

File: backend/schemas/artifacts.py
from datetime import datetime
from typing import Literal

from pydantic import BaseModel, ConfigDict


class ArtifactRead(BaseModel):
    model_config = ConfigDict(from_attributes=True, extra="forbid")

    id: str
    kind: Literal["image", "output", "checkpoint", "dataset"]
    path_type: Literal["file", "directory"]
    display_name: str | None = None
    created_at: datetime
    content_url: str
